zmq.hpp

```diff
@@ -1058,18 +1058,17 @@ namespace zmq
 
         void add (zmq::socket_t &socket, short events, handler_t handler)
         {
-            if (!socket)
-                throw error_t ();
-            handler_t *handler_ptr = nullptr;
-            /// \todo is it sensible to allow handler to be empty? doesn't this lead to an error when the event is signalled? (perhaps it should already lead to an error in zmq_poller_add then)
-            if (handler) {
-                auto emplace_res = handlers.emplace (socket.ptr, std::move (handler));
-                handler_ptr = &emplace_res.first->second;
-            }
-            if (0 == zmq_poller_add (poller_ptr, socket.ptr, handler_ptr, events)) {
+            auto it = std::end (handlers);
+            auto inserted = false;
+            if (handler)
```

Comment:
The comment was removed, but it is still applicable.

Reply:
Sorry, should have explained that. This comment was only confusing as it is valid to pass nullptr user data to zmq_poller_add. libzmq does not care what user passes there, it only passively handles it back on wait. Similarly to actors in czmq

Reply:
It is valid for libzmq/zmq_poller_add, but it is not sensible in the way cppzmq uses it. In wait, a no-op results if handler was nullptr here.

Reply:
if you are referring to if we should allow empty handler at all then this is a separate discussion.
You should not create //todo comments for that because that might indicate that something was agreed (where?) and waiting to be done. If you think if something can be improved why not create an issue open for discussion, or provide pull request..
In my opinion we should allow empty handlers, it is a valid use case where user is interested in wait signalling only by returning and no-op.

Reply:
My original comment was indeed mistaken. When I wrote it, I thought that it would lead to some error later on, but this is not the case. So I am fine to remove it.

If you used a poller with a single socket and an empty handler, it might make sense. But it is somewhat misaligned with the design of poller_t, which manages and calls back into the handlers. I would still tend to disallow is, but I am also fine to leave it as it is. It would be good to have a test that uses it this way (i.e. not only for adding an empty handler, but also polling afterwards).